### DISCUSSION TITLE
send_velocity_world_setpoint(), yawrate is in Body or World Frame ?

### DISCUSSION THREAD

**Discussion by imamim** [bitcraze/discussions#650] [Q&A]
Hello, I want to speed up crazyflie using the send_velocity_world_setpoint() function using the angular velocity vector in the world frame that I obtained for a trajectory tracking.

What frame should the yawrate value given to the function be on? Does this mean the world-z axis value of the angular velocity value or the body-z axis? I'm guessing it's probably on the body frame. But I wanted to make sure as I don't know how the transactions are done inside.

Thank you for your answer.

**Comment by krichardsson** [comment]
It is the body frame

The setpoint is [decoded here](https://github.com/bitcraze/crazyflie-firmware/blob/26b016b42b4eff6e07fcaa597c35925e5f9c7242/src/modules/src/crtp_commander_generic.c#L98-L115) if you want to dig deeper in the code

**Comment by imamim** [comment]
Thank you for your answer, that's okey for me
